js/expiring_messages.js

```diff
@@ -31,8 +31,13 @@
             console.log('next message expires', new Date(expires_at).toISOString());
 
             var wait = expires_at - Date.now();
+
+            // In the past
             if (wait < 0) { wait = 0; }
 
+            // Too far in the future, since it's limited to a 32-bit value
+            if (wait > 2147483647) { wait = 2147483647; }
```

Comment:
Good to know. There are a couple events which force a check and re-schedule - any new disappearing message, any detection of time travel, a previously-scheduled timer firing. I think we're in good shape.